packages/makeup-floating-label/index.js

```diff
@@ -11,27 +11,56 @@ function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _d
 var defaultOptions = {
   labelElementAnimateModifier: 'floating-label__label--animate',
   labelElementInlineModifier: 'floating-label__label--inline',
-  textboxElementBackgroundRGB: 'rgb(255, 255, 255)'
+  labelElementFocusModifier: 'floating-label__label--focus',
+  labelElementInvalidModifier: 'floating-label__label--invalid',
+  labelElementDisabledModifier: 'floating-label__label--disabled',
+  textboxElementBackgroundRGB: ['rgb(255, 255, 255)', 'rgb(245, 245, 245)', 'rgb(230, 32, 72)', 'rgb(254, 245, 246)']
 };
 
+function onMutation() {
+  if (isInvalid(this.textboxEl)) {
+    this.labelEl.classList.add(this.options.labelElementInvalidModifier);
+  } else {
+    this.labelEl.classList.remove(this.options.labelElementInvalidModifier);
+  }
+
+  if (isDisabled(this.textboxEl)) {
+    this.labelEl.classList.add(this.options.labelElementDisabledModifier);
+  } else {
+    this.labelEl.classList.remove(this.options.labelElementDisabledModifier);
+  }
+}
+
 function hasValue(input) {
   return input.value.length > 0;
 }
 
+function isDisabled(input) {
+  return input.hasAttribute('disabled');
+}
+
+function isInvalid(input) {
+  return input.hasAttribute('aria-invalid') && input.getAttribute('aria-invalid') === 'true';
+}
+
 function isAutofilled(input, color) {
   // check for computed background color because of Chrome autofill bug
   // https://stackoverflow.com/questions/35049555/chrome-autofill-autocomplete-no-value-for-password/35783761#35783761
-  return getComputedStyle(input).backgroundColor !== color;
+  var bgColor = getComputedStyle(input).backgroundColor;
+  return Array.isArray(color) ? !color.includes(bgColor) : bgColor !== color;
 }
 
 function _onBlur() {
   if (!hasValue(this.textboxEl)) {
     this.labelEl.classList.add(this.options.labelElementInlineModifier);
   }
+
+  this.labelEl.classList.remove(this.options.labelElementFocusModifier);
 }
 
 function _onFocus() {
   this.labelEl.classList.add(this.options.labelElementAnimateModifier);
+  this.labelEl.classList.add(this.options.labelElementFocusModifier);
   this.labelEl.classList.remove(this.options.labelElementInlineModifier);
 }
 
```

Comment:
So this isn't related to your changes necessarily, but why are we exporting an IIFE on line 67? Why not just export a class?

Reply:
This index.js file is the compiled output from Babel.

Reply:
that makes more sense thanks for the clarification